pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -133,10 +134,36 @@ func (r *ReconcileConstraintTemplate) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	instance.Status.Errors = []*v1alpha1.CreateCRDError{}
 	crd, err := r.opa.CreateCRD(context.Background(), instance)
 	if err != nil {
+		var createErr *v1alpha1.CreateCRDError
+		if parseErrs, ok := err.(ast.Errors); ok {
+			for i := 0; i < len(parseErrs); i++ {
+				createErr = &v1alpha1.CreateCRDError{Code: parseErrs[i].Code, Message: parseErrs[i].Message, Location: parseErrs[i].Location.String()}
+				instance.Status.Errors = append(instance.Status.Errors, createErr)
+			}
+		} else {
+			createErr = &v1alpha1.CreateCRDError{Code: "unknown_error", Message: err.Error(), Location: ""}
+			instance.Status.Errors = append(instance.Status.Errors, createErr)
+		}
+
+		if updateErr := r.Update(context.Background(), instance); updateErr != nil {
+			log.Error(updateErr, "update error", updateErr)
+			return reconcile.Result{Requeue: true}, nil
+		}
 		return reconcile.Result{}, err
 	}
+
+	if len(instance.Status.Errors) > 0 {
```

Comment:
Why are we updating the resource twice?

Or is this meant to wipe previous errors if err == nil and there are pre-existing errors? If so, we need to set `instance.Status.Errors = nil` before updating. We would also need to move this into an else-if block.

Reply:
I removed the first update, it was redundant.